File: examples/pyapp/store.py
import threading
import uuid
from datetime import datetime, timezone


class Task:
    def __init__(self, title: str, description: str = ""):
        self.id = str(uuid.uuid4())[:8]
        self.title = title
        self.description = description
        self.done = False
        self.created_at = datetime.now(timezone.utc).isoformat()
        self.updated_at = self.created_at

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "title": self.title,
            "description": self.description,
            "done": self.done,
            "created_at": self.created_at,
            "updated_at": self.updated_at,
        }


class Store:
    def __init__(self):
        self._tasks: dict[str, Task] = {}
        self._lock = threading.Lock()

    def list(self) -> list[dict]:
        with self._lock:
            return [t.to_dict() for t in self._tasks.values()]

    def create(self, title: str, description: str = "") -> dict:
        task = Task(title, description)
        with self._lock:
            self._tasks[task.id] = task
        return task.to_dict()

    def get(self, task_id: str) -> dict | None:
        with self._lock:
            t = self._tasks.get(task_id)
            return t.to_dict() if t else None

    def update(self, task_id: str, title: str, description: str) -> dict | None:
        with self._lock:
            t = self._tasks.get(task_id)
            if not t:
                return None
            t.title = title
            t.description = description
            t.updated_at = datetime.now(timezone.utc).isoformat()
            return t.to_dict()

    def complete(self, task_id: str) -> dict | None:
        with self._lock:
            t = self._tasks.get(task_id)
            if not t:
                return None
            t.done = True
            t.updated_at = datetime.now(timezone.utc).isoformat()
            return t.to_dict()

    def delete(self, task_id: str) -> bool:
        with self._lock:
            if task_id not in self._tasks:
                return False
            del self._tasks[task_id]
            return True

    def count(self) -> int:
        with self._lock:
            return len(self._tasks)

File: examples/webapp/internal/store/store.go
package store

import (
	"fmt"
	"sync"
	"time"
)

type Task struct {
	ID          string    `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Done        bool      `json:"done"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type Store struct {
	mu    sync.RWMutex
	tasks map[string]Task
	next  int
}

func New() *Store {
	return &Store{tasks: make(map[string]Task)}
}

func (s *Store) Create(title, description string) Task {
	s.mu.Lock()
	defer s.mu.Unlock()
	s.next++
	t := Task{
		ID:          fmt.Sprintf("%d", s.next),
		Title:       title,
		Description: description,
		CreatedAt:   time.Now(),
		UpdatedAt:   time.Now(),
	}
	s.tasks[t.ID] = t
	return t
}

func (s *Store) Get(id string) (Task, bool) {
	s.mu.RLock()
	defer s.mu.RUnlock()
	t, ok := s.tasks[id]
	return t, ok
}

func (s *Store) List() []Task {
	s.mu.RLock()
	defer s.mu.RUnlock()
	tasks := make([]Task, 0, len(s.tasks))
	for _, t := range s.tasks {
		tasks = append(tasks, t)
	}
	return tasks
}

func (s *Store) Update(id, title, description string) (Task, bool) {
	s.mu.Lock()
	defer s.mu.Unlock()
	t, ok := s.tasks[id]
	if !ok {
		return Task{}, false
	}
	t.Title = title
	t.Description = description
	t.UpdatedAt = time.Now()
	s.tasks[id] = t
	return t, true
}

func (s *Store) Complete(id string) (Task, bool) {
	s.mu.Lock()
	defer s.mu.Unlock()
	t, ok := s.tasks[id]
	if !ok {
		return Task{}, false
	}
	t.Done = true
	t.UpdatedAt = time.Now()
	s.tasks[id] = t
	return t, true
}

func (s *Store) Delete(id string) bool {
	s.mu.Lock()
	defer s.mu.Unlock()
	_, ok := s.tasks[id]
	if ok {
		delete(s.tasks, id)
	}
	return ok
}

func (s *Store) Count() int {
	s.mu.RLock()
	defer s.mu.RUnlock()
	return len(s.tasks)
}
